### ISSUE TITLE
Accessibility: Include alt attribute for amp-img component (and amp-anim, amp-slides, amp-carousel?)

### ISSUE THREAD

**Issue by alishalisha** [ampproject/amphtml#176]
I'd like to suggest this as an official part of the AMP spec to help those who use screen readers and have other disabilities.

Example:
```html
<amp-img alt="Description of the image."... /></amp-img>
```

This same idea could also be extended to the slides, anim, and carousel components.

Further reading on ALT text and accessibility here: http://a11yproject.com/posts/alt-text/

**Comment by stevefaulkner** [comment]
>The alt attributes definitely work. The question would be whether they should be mandatory, I guess.

>I don't have a strong opinion on that.

is the amp-img element is a wrapper for an <img> element in the DOM?

**Comment by dvoytenko** [comment]
@stevefaulkner Yes, they are. And we do support "alt" and all of the "aria" attributes as well.

**Comment by dvoytenko** [comment]
I think we will also be getting a lot more aggressive with reporting accessibility issues, such as missing "alt" or "aria"/"role" attributes back to authors. However, so far we've planned to do so as warnings, not errors.

**Comment by Gregable** [comment]
My 2c on the mandatory question is that requiring an alt attribute will simply have lots of authors add alt='' or alt=' ' or whatever is required to make validation pass. There is no way to validate that the value of the attribute is improving accessibility. Given that, I don't see the logic in making the attribute itself mandatory. I could see having the validation emit some light warnings, but it should be clear that these aren't the type of warning that will become errors later (e.g. not deprecation warnings).
